tiledb/cf/engines/netcdf4_engine.py

```diff
@@ -28,8 +29,27 @@
 COORDINATE_SUFFIX = ".data"
 
 
+class NetCDFDimConverter(ABC):
+    @abstractmethod
+    def get_values(
+        self, netcdf_group: netCDF4.Dataset, sparse: bool
+    ) -> Union[np.ndarray, slice]:
+        """Returns the values of the NetCDF dimension that is being copied, or None if
+        the dimension is of size 0.
+
+        Parameters:
+            netcdf_group: NetCDF group to get the dimension values from.
+            sparse: ``True`` if copying into a sparse array and ``False`` if copying
+                into a dense array.
+
+        Returns:
+            The coordinates needed for querying the create TileDB dimension in the form
+                of a numpy array if sparse is ``True`` and a slice otherwise.
+        """
+
+
 @dataclass
-class NetCDFDimensionConverter(SharedDim):
+class NetCDFDimToDimConverter(SharedDim, NetCDFDimConverter):
     """Data for converting from a NetCDF dimension to a TileDB dimension.
 
     Parameters:
```

Comment:
The parameters and attributes are the same in the description, is this correct?

Reply:
This is correct for dataclasses in general. Maybe it's better to add just one or the other. However, if I change that, I would want to change it for all the dataclasses, and so I would want it to be a separate PR.